remote_engine_client/src/client.rs

```diff
@@ -15,27 +17,40 @@ use ceresdbproto::{
 use common_types::{
     projected_schema::ProjectedSchema, record_batch::RecordBatch, schema::RecordSchema,
 };
-use common_util::error::BoxError;
-use futures::{Stream, StreamExt};
+use common_util::{error::BoxError, runtime::Runtime};
+use futures::{future::join_all, Stream, StreamExt};
 use router::RouterRef;
 use snafu::{ensure, OptionExt, ResultExt};
 use table_engine::{
-    remote::model::{GetTableInfoRequest, ReadRequest, TableIdentifier, TableInfo, WriteRequest},
+    remote::model::{
+        GetTableInfoRequest, ReadRequest, TableIdentifier, TableInfo, WriteBatchRequest,
+        WriteBatchResult, WriteRequest,
+    },
     table::{SchemaId, TableId},
 };
 use tonic::{transport::Channel, Request, Streaming};
 
 use crate::{cached_router::CachedRouter, config::Config, error::*, status_code};
 
+struct WriteBatchContext {
+    table_idents: Vec<TableIdentifier>,
```

Comment:
I guess this fields is unnecessary if it is just for error message.

Reply:
It is for evict, too.